File: vripper-server/src/main/java/tn/mnlr/vripper/web/restendpoints/EventLogRestEndpoint.java
package tn.mnlr.vripper.web.restendpoints;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;
import tn.mnlr.vripper.jpa.repositories.ILogEventRepository;

@Slf4j
@RestController
@CrossOrigin(value = "*")
public class EventLogRestEndpoint {

  private final ILogEventRepository eventRepository;

  public EventLogRestEndpoint(ILogEventRepository eventRepository) {
    this.eventRepository = eventRepository;
  }

  @GetMapping("/events/clear")
  @ResponseStatus(value = HttpStatus.OK)
  public void clear() {
    eventRepository.deleteAll();
  }
}

File: vripper-web/src/main/kotlin/me/mnlr/vripper/web/restendpoints/EventLogRestEndpoint.kt
package me.mnlr.vripper.web.restendpoints

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import me.mnlr.vripper.repositories.LogEventRepository
import org.springframework.web.bind.annotation.RequestMapping

@RestController
@RequestMapping("/api")
class EventLogRestEndpoint(private val logEventRepository: LogEventRepository) {
    @GetMapping("/events/clear")
    @ResponseStatus(value = HttpStatus.OK)
    fun clear() {
        logEventRepository.deleteAll()
    }
}